documentation/library_guide/dynamic_selection_api/functions.rst

```diff
@@ -348,68 +200,63 @@ The output from this example is::
-The use of ``single_task`` is for syntactic demonstration 
-purposes only; any valid command group or series of command groups can be 
-submitted to the selected queue.
+This function is useful when you want to handle the case where no resources
+are immediately available without blocking.
 
-.. code::  cpp
+.. code:: cpp
 
   #include <oneapi/dpl/dynamic_selection>
   #include <sycl/sycl.hpp>
   #include <iostream>
-  
+
   namespace ex = oneapi::dpl::experimental;
-  
+
   int main() {
-    ex::round_robin_policy p{ { sycl::queue{ sycl::cpu_selector_v },  
+    ex::round_robin_policy p{ { sycl::queue{ sycl::cpu_selector_v },
                                 sycl::queue{ sycl::gpu_selector_v } } };
-  
+
     for (int i = 0; i < 4; ++i) {
-      auto s = ex::select(p);
-      ex::submit_and_wait(/* selection object */ s,  
-                          /* user function */
-                          [](sycl::queue q, /* any additional args... */ int j) {
-                             std::cout << "(j == " << j << "): submit to " 
-                                       << ((q.get_device().is_gpu()) ? "gpu\n" : "cpu\n");
-                             auto e = q.single_task([]() { /* do some work */ }); 
-                             return e; /* MUST return sycl::event */
-                          },
-                          /* any additional args... */ i);  
-      std::cout << "(i == " << i << "): submission done\n"; 
+      auto maybe_done = ex::try_submit(p,
+                                       [](sycl::queue q, int j) {
+                                         std::cout << "(j == " << j << "): submit to "
+                                                   << ((q.get_device().is_gpu()) ? "gpu\n" : "cpu\n");
+                                         return q.single_task([]() { /* work */ });
+                                       },
+                                       i);
+
+      if (maybe_done.has_value()) {
+        std::cout << "(i == " << i << "): submission succeeded\n";
+        ex::wait(maybe_done.value());
+      } else {
+        std::cout << "(i == " << i << "): no resource available, trying alternative work\n";
+        // Could retry, do other work, or handle the failure differently
+      }
     }
   }
 
+**Note:** For most policies like ``round_robin_policy``, resources are always
+available for a well formed policy with existing resources.
+``try_submit`` is most useful with policies that may temporarily have no
+available resources, such as a custom policy which has hard capacity limits
+on specific devices.
 
-The output from this example is::
-
-  (j == 0): submit to cpu
-  (i == 0): submission done
-  (j == 1): submit to gpu
-  (i == 1): submission done
-  (j == 2): submit to cpu
-  (i == 2): submission done
-  (j == 3): submit to gpu
-  (i == 3): submission done
+The ``submit`` function uses ``try_submit`` internally and retries
+with backoff when no resource is available.
```

Comment:
Is this important for the user documentation? Seems like an exposition of implementation details.

Reply:
removed